### PR TITLE
Add validation for deprecated input variables

### PR DESCRIPTION
**Pull request by rohitramu**

### CODE REVIEW

File: modules/network/vpc/variables.tf [hunk 1]
```diff
@@ -118,44 +122,52 @@ variable "primary_subnetwork" {
   type        = map(string)
   default     = null
   validation {
-    condition     = var.primary_subnetwork == null || can(var.primary_subnetwork["subnet_name"])
+    condition     = var.primary_subnetwork == null ? true : can(var.primary_subnetwork["subnet_name"])
```
Comment (mr0re1) [inline modules/network/vpc/variables.tf:125]: Why this change? Aren't both of those lines are semantically identical?

File: modules/network/vpc/variables.tf [hunk 2]
```diff
@@ -118,44 +122,52 @@ variable "primary_subnetwork" {
...
     error_message = "Primary subnetwork must define \"subnet_name\"."
   }
   validation {
-    condition     = var.primary_subnetwork == null || can(var.primary_subnetwork["subnet_region"])
```
Comment (mr0re1) [inline modules/network/vpc/variables.tf:125]: I guess all those old validation blocks can be simply removed.

File: modules/network/vpc/variables.tf (hunk 1 above)
Comment (rohitramu) [reply modules/network/vpc/variables.tf:125]: Unfortunately, HCL doesn't use short-circuiting in conditionals: https://github.com/hashicorp/terraform/issues/24128#issuecomment-1155241478

However, it does for ternary statements (i.e. `<condition> ? <true_action> : <false_action>`).

File: tests.log [hunk 3]
```diff
@@ -0,0 +1,111 @@
+**************** vetting go code **********************
```
Comment (mr0re1) [inline tests.log:1]: Please clean up added files. `tests.log` & `~/Documents`

Comment (rohitramu) [reply tests.log:1]: I have no idea how files outside the repo got added!  Thanks for flagging this, I'll clean it up.

File: modules/network/vpc/variables.tf (hunk 1 above)
Comment (mr0re1) [reply modules/network/vpc/variables.tf:125]: How does lack of short-circuiting changes anything here?
IIUC `can(expr)` will return `false` if `expr` throws an error, so `var.primary_subnetwork == null` shouldn't be a problem here.

Comment (rohitramu) [reply modules/network/vpc/variables.tf:125]: That's a good point, I'll revert the changes which involve `can()`.

File: modules/network/vpc/variables.tf (hunk 2 above)
Comment (rohitramu) [reply modules/network/vpc/variables.tf:125]: Done!

File: modules/network/vpc/variables.tf [hunk 4]
```diff
@@ -118,44 +122,24 @@ variable "primary_subnetwork" {
...
-      for s in var.additional_subnetworks : can(s["subnet_region"])
-    ])
-    error_message = "All additional subnetworks must define \"subnet_region\"."
-  }
+  default     = null
   validation {
-    condition     = alltrue([for s in var.additional_subnetworks : can(s["new_bits"])])
-    error_message = "All additional subnetworks must define \"new_bits\"."
+    condition     = var.additional_subnetworks == null
+    error_message = "additional_subnetworks is deprecated."
```
Comment (mr0re1) [inline modules/network/vpc/variables.tf:136]: Add `Please see https://goo.gle/hpc-toolkit-vpc-deprecation for migration instructions.`

**Review by mr0re1 — APPROVED** [review]

File: modules/network/vpc/variables.tf (hunk 4 above)
Comment (rohitramu) [reply modules/network/vpc/variables.tf]: Good catch - I've added it now.
